### PR TITLE
Add quote for change items/reactivate order

### PR DESCRIPTION
**Pull request by lexerom**
## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Comment by lexerom** [comment]
We need to use discriminator here for Quote.action (ReactivateQuote, ChangeQuote, CreateQuote)

File: openapi/components/schemas/Quote.yaml [hunk 1]
```diff
@@ -1,357 +1,10 @@
...
-        type: object
-      order:
-        type: object
+discriminator:
+  propertyName: action
+  mapping:
+    create: ./QuoteCreateOrder.yaml
+    change: ./QuoteChangeOrder.yaml
+    reactivate: ./QuoteReactivateOrder.yaml
+anyOf:
```
Comment (justnero) [inline openapi/components/schemas/Quote.yaml:7]: Discriminator property is required and mutually exclusive
```suggestion
oneOf:
```

File: openapi/components/schemas/StorefrontQuote.yaml [hunk 2]
```diff
@@ -1,331 +1,10 @@
...
-        type: object
-      website:
-        type: object
+discriminator:
+  propertyName: action
+  mapping:
+    create: ./StorefrontQuoteCreateOrder.yaml
+    change: ./StorefrontQuoteChangeOrder.yaml
+    reactivate: ./StorefrontQuoteReactivateOrder.yaml
+anyOf:
```
Comment (justnero) [inline openapi/components/schemas/StorefrontQuote.yaml:7]:
```suggestion
oneOf:
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote for change order items.
+      On accept quote interim invoice is issued and linked to `invoiceId`.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:26]: suggestion: for readability
```suggestion
      When the accept quote action is used, an interim invoice is issued and linked to `invoiceId`.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+    items:
+      type: object
+      properties:
+        href:
+          description: Link URL.
+          type: string
+        rel:
+          description: |-
+            Type of link. If quote is accepted, a customer is redirected to `invoicePaymentFormUrl`
+            to pay the initial invoice using one of the methods which are available to the customer.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:466]: suggestion: for readability
```suggestion
            Type of link. If the quote is accepted, a customer is redirected to the  `invoicePaymentFormUrl` value
            to pay the initial invoice using one of the methods which are available to the customer.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+        customer: Quote can be accepted by customer.
+        signature: Quote can be accepted if merchant received the signed quote from the customer.
+        payment: Related invoice must be paid in full.
+        partial-payment: Related invoice must be paid in full or partially paid.
+    example: ['customer', 'payment']
+    default: ['customer']
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:51]: suggestion: to add more detail
I found this description difficult to understand (I'm not sure if my suggestion changes the meaning)

```suggestion
    description: |-
      Conditions that must be completed to fulfill the order. 
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+      The related order will have `pending` status.
+      Once the invoice is paid, the order becomes active on `settings.reactivate.effectiveTime` or immediately if `settings.reactivate.effectiveTime` is omitted.
+    type: string
+    enum:
+      - reactivate
+    example: reactivate
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:33]: suggestion: for readability 
```suggestion
    description: Conditions that must be met for the quote to be accepted.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote for churned/canceled order reactivation.
+      On accept quote new invoice is issued to be and linked to `invoiceId`. Quote items are applied to the order.
+      The related order will have `pending` status.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:24]: suggestion: for readability 
```suggestion
      Action of the quote for churned or canceled order reactivation.
      When the accept quote action is used,
      a new invoice is issued to be and linked to the `invoiceId`. Quote items are applied to the order.
      The related order will have the `pending` status.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+      After the invoice is paid changed items are applied to an order with `orderId`.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:27]: suggestion: I think we don't need to state 'order with `orderId`'
all orders have an orderId (not sure if I'm misunderstanding something here)
```suggestion
      After the invoice is paid, 
      changed items are applied to the related order.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+        customer: Quote can be accepted by customer.
+        signature: Quote can be accepted if merchant received the signed quote from the customer.
+        payment: Related invoice must be paid in full.
+        partial-payment: Related invoice must be paid in full or partially paid.
+    example: ['customer', 'payment']
+    default: ['customer']
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:53]: suggestion: to add more detail
I found this description difficult to understand (I'm not sure if my suggestion changes the meaning)

```suggestion
    description: Conditions that must be met to fulfill the order. 
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+      On accept quote interim invoice is issued and linked to `invoiceId`.
+      After the invoice is paid changed items are applied to an order with `orderId`.
+    type: string
+    enum:
+      - change
+    example: change
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:35]: suggestion: for readability 
```suggestion
    description: Conditions that must be met for the quote to be accepted.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:372]: suggestion: for readability ([noun stack](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#noun-strings-or-noun-stacks))
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 12]
```diff
@@ -0,0 +1,468 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote.
+      On accept quote new order is created. `orderId` and `invoiceId` are filled with new generated values from the order.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:25]:
```suggestion
      When the accept quote action is used, a new order is created. 
      `orderId` and `invoiceId` are filled with new generated values from the order.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 13]
```diff
@@ -0,0 +1,468 @@
...
+    type: string
+    enum:
+      - create
+    example: create
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:33]:
```suggestion
    description: If conditions are fulfilled the quote is automatically accepted.
```

suggestion: for readability 
```suggestion
    description: Conditions that must be met for the quote to be accepted.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 14]
```diff
@@ -0,0 +1,468 @@
...
+        customer: Quote can be accepted by customer.
+        signature: Quote can be accepted if merchant received the signed quote from the customer.
+        payment: Related invoice must be paid in full.
+        partial-payment: Related invoice must be paid in full or partially paid.
+    example: ['customer', 'payment']
+    default: ['customer']
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:51]: suggestion: to add more detail
I found this description difficult to understand (I'm not sure if my suggestion changes the meaning)

```suggestion
    description: Conditions that must be met to fulfill the order. 
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 15]
```diff
@@ -0,0 +1,468 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:354]:
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 16]
```diff
@@ -0,0 +1,493 @@
...
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote for churned/canceled order reactivation.
+      On accept quote new invoice is issued to be and linked to `invoiceId`. Quote items are applied to the order.
+      The related order will have `pending` status.
+      Once the invoice is paid, the order becomes active on `settings.reactivate.effectiveTime` or immediately if `settings.reactivate.effectiveTime` is omitted.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:28]: suggestion: for readability 
```suggestion
      Action of the quote for churned or canceled order reactivation.
      When the accept quote action is used,
      a new invoice is issued to be and linked to the `invoiceId`. 
      Quote items are applied to the order.
      The related order will have the `pending` status.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 17]
```diff
@@ -0,0 +1,493 @@
...
+    type: string
+    enum:
+      - reactivate
+    example: reactivate
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:36]: suggestion: for readability 
```suggestion
    description: Conditions that must be met for the quote to be accepted.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 18]
```diff
@@ -0,0 +1,493 @@
...
+        customer: Quote can be accepted by customer.
+        signature: Quote can be accepted if merchant received the signed quote from the customer.
+        payment: Related invoice must be paid in full.
+        partial-payment: Related invoice must be paid in full or partially paid.
+    example: ['customer', 'payment']
+    default: ['customer']
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:54]: suggestion: to add more detail
I found this description difficult to understand (I'm not sure if my suggestion changes the meaning)

```suggestion
    description: Conditions that must be met to fulfill the order. 
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 19]
```diff
@@ -0,0 +1,474 @@
...
+        customer: Quote can be accepted by customer.
+        signature: Quote can be accepted if merchant received the signed quote from the customer.
+        payment: Related invoice must be paid in full.
+        partial-payment: Related invoice must be paid in full or partially paid.
+    example: ['customer', 'payment']
+    default: ['customer']
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:50]: suggestion: to add more detail
I found this description difficult to understand (I'm not sure if my suggestion changes the meaning)

```suggestion
    description: Conditions that must be met to fulfill the order. 
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 20]
```diff
@@ -0,0 +1,474 @@
...
+    items:
+      type: object
+      properties:
+        href:
+          description: Link URL.
+          type: string
+        rel:
+          description: |-
+            Type of link. If quote is accepted, a customer is redirected to `invoicePaymentFormUrl`
+            to pay the initial invoice using one of the methods which are available to the customer.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:459]:
```suggestion
            Type of link. 
            If the quote is accepted, the customer is redirected to `invoicePaymentFormUrl`
            to pay the initial invoice using one of the methods which are available to the customer.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 21]
```diff
@@ -0,0 +1,455 @@
...
+    items:
+      type: object
+      properties:
+        href:
+          description: Link URL.
+          type: string
+        rel:
+          description: |-
+            Type of link. If quote is accepted, a customer is redirected to `invoicePaymentFormUrl`
+            to pay the initial invoice using one of the methods which are available to the customer.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:440]:
```suggestion
            Type of link.
            If the quote is accepted, the customer is redirected to `invoicePaymentFormUrl`
            to pay the initial invoice using one of the methods which are available to the customer.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 22]
```diff
@@ -0,0 +1,493 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:379]:
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 23]
```diff
@@ -0,0 +1,474 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:369]: suggestion: for readability
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 24]
```diff
@@ -0,0 +1,455 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:350]: suggestion: for readability
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 25]
```diff
@@ -0,0 +1,481 @@
...
+                - 'number'
+                - 'null'
+              format: double
+            trialPeriod:
+              description: Date interval of the invoice item trial.
+              type:
+                - 'string'
+                - 'null'
+            taxAmount:
+              description: Invoice item tax.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:376]: suggestion: for readability
```suggestion
              description: Tax amount of the invoice item.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 26]
```diff
@@ -0,0 +1,474 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote for change order items.
+      On accept quote interim invoice is issued and linked to `invoiceId`.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:23]:
```suggestion
      When the accept quote action is used,
      and interim invoice is issued and linked to `invoiceId`.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 27]
```diff
@@ -0,0 +1,474 @@
...
+      After the invoice is paid changed items are applied to an order with `orderId`.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:24]:
```suggestion
      After the invoice is paid, changed items are applied to the order.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 28]
```diff
@@ -0,0 +1,455 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote.
+      On accept quote new order is created. `orderId` and `invoiceId` are filled with new generated values from the order.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:23]:
```suggestion
      When the accept quote action is used,
      a new order is created. 
      `orderId` and `invoiceId` are filled with new generated values from the order.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 29]
```diff
@@ -0,0 +1,474 @@
...
+    type: string
+    enum:
+      - change
+    example: change
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:32]: suggestion: for readability
```suggestion
    description: Conditions that must be met for the quote to be accepted.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 30]
```diff
@@ -0,0 +1,486 @@
...
+  acceptanceFulfillment:
+    type: array
+    readOnly: true
+    description: Conditions with fulfillment flag.
+    items:
+      type: object
+      properties:
+        condition:
+          type: string
+          enum:
```
Comment (danio1024) [inline openapi/components/schemas/QuoteChangeOrder.yaml:59]: Add `merchant` to signify that merchant manually accepted.

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 31]
```diff
@@ -20,44 +16,38 @@ properties:
   action:
     description: |-
       Action of the quote for change order items.
-      On accept quote interim invoice is issued and linked to `invoiceId`.
-      After the invoice is paid changed items are applied to an order with `orderId`.
+      On acceptance quote items are applied to the related order.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:19]: suggestion: for readability
```suggestion
      When a quote is accepted, quote items are applied to the related order.
```

Comment (lexerom) [reply openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:19]: Thanks, will change everywhere.

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 32]
```diff
@@ -0,0 +1,482 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:423]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 33]
```diff
@@ -0,0 +1,482 @@
...
+    type: string
+    enum:
+      - reset
+      - retain
+  prorated:
+    description: |-
+      Specifies whether to give a pro rata credit for the amount of time remaining between the `effectiveTime` and the end of the current period.
+
+      In addition, if the `renewalTime` is retained, by setting the `renewalPolicy` to `retain`,
+      a pro rata debit occurs for the amount between the `effectiveTime` and the `renewalTime` as a percentage of the normal period size.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteChangeOrder.yaml:212]: suggestion:
```suggestion
      a pro rata debit occurs for the amount between the `effectiveTime` and the `renewalTime` as a percentage of the normal period length.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 34]
```diff
@@ -0,0 +1,467 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote.
+      When a quote is accepted, a new order is created. `orderId` is filled with new generated value from the order.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:25]: suggestion: for readability 
```suggestion
      When a quote is accepted, a new order is created. 
      The `orderId` value is filled with a new generated value from the order.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 35]
```diff
@@ -0,0 +1,467 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteCreateOrder.yaml:408]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 36]
```diff
@@ -0,0 +1,469 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:422]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 37]
```diff
@@ -0,0 +1,446 @@
...
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote.
+      When a quote is accepted, a new order is created. `orderId` is filled with new generated value from the order.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:19]: suggestion: for readability 
```suggestion
      When a quote is accepted, a new order is created. 
      The `orderId` value is filled with a new generated value from the order.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 38]
```diff
@@ -0,0 +1,446 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:401]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 39]
```diff
@@ -0,0 +1,463 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:416]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 40]
```diff
@@ -0,0 +1,463 @@
...
+    type: string
+    enum:
+      - reset
+      - retain
+  prorated:
+    description: |-
+      Specifies whether to give a pro rata credit for the amount of time remaining between the `effectiveTime` and the end of the current period.
+
+      In addition, if the `renewalTime` is retained, by setting the `renewalPolicy` to `retain`,
+      a pro rata debit occurs for the amount between the `effectiveTime` and the `renewalTime` as a percentage of the normal period size.
```
Comment (noon-dawg) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:205]: suggestion:
```suggestion
      a pro rata debit occurs for the amount between the `effectiveTime` and the `renewalTime` as a percentage of the normal period length.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 41]
```diff
@@ -0,0 +1,488 @@
...
+  type:
+    description: Specifies the type of the quote. A quote can be a subscription or a one-time purchase.
+    type: string
+    readOnly: true
+    enum:
+      - subscription-order
+      - one-time-order
+  action:
+    description: |-
+      Action of the quote for churned/canceled order reactivation.
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:25]: suggestion: for readability we should avoid using '/' 
```suggestion
      Action of the quote for churned or canceled order reactivation.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 42]
```diff
@@ -0,0 +1,488 @@
...
+      - 'string'
+      - 'null'
+    readOnly: true
+    format: date-time
+  redirectUrl:
+    description: URL to redirect the customer to when a quote is rejected. The default value is the website URL.
+    type: string
+    format: uri
+  signature:
+    type: object
```
Comment (noon-dawg) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:429]: nitpick: No top level description rendered in the docs for `signature
Screenshot 2024-01-26 at 15 21 27

Suggested description:

```suggestion
    type: object
    description: Written signature and printed organization name.
```

File: openapi/components/schemas/QuoteChangeOrder.yaml [hunk 43]
```diff
@@ -0,0 +1,483 @@
...
+      Action of the quote for change order items.
+      When a quote is accepted, quote items are applied to the related order.
+    type: string
+    enum:
+      - change
+    example: change
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/QuoteChangeOrder.yaml:34]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

File: openapi/components/schemas/QuoteCreateOrder.yaml [hunk 44]
```diff
@@ -0,0 +1,469 @@
...
+      When a quote is accepted, a new order is created. 
+      The `orderId` value is filled with a new generated value from the order.
+    type: string
+    enum:
+      - create
+    example: create
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/QuoteCreateOrder.yaml:34]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

File: openapi/components/schemas/QuoteReactivateOrder.yaml [hunk 45]
```diff
@@ -0,0 +1,489 @@
...
+      Action of the quote for churned or canceled order reactivation.
+      When a quote is accepted, quote items are applied to the order and order is reactivated.
+    type: string
+    enum:
+      - reactivate
+    example: reactivate
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/QuoteReactivateOrder.yaml:34]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

File: openapi/components/schemas/StorefrontQuoteChangeOrder.yaml [hunk 46]
```diff
@@ -0,0 +1,464 @@
...
+      Action of the quote for change order items.
+      When a quote is accepted, quote items are applied to the related order.
+    type: string
+    enum:
+      - change
+    example: change
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/StorefrontQuoteChangeOrder.yaml:27]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

File: openapi/components/schemas/StorefrontQuoteCreateOrder.yaml [hunk 47]
```diff
@@ -0,0 +1,448 @@
...
+      When a quote is accepted, a new order is created. 
+      The `orderId` value is filled with a new generated value from the order.
+    type: string
+    enum:
+      - create
+    example: create
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/StorefrontQuoteCreateOrder.yaml:28]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

File: openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml [hunk 48]
```diff
@@ -0,0 +1,470 @@
...
+      Action of the quote for churned/canceled order reactivation.
+      When a quote is accepted, quote items are applied to the order and order is reactivated.
+    type: string
+    enum:
+      - reactivate
+    example: reactivate
+  acceptanceConditions:
+    type: array
+    writeOnly: true
+    description: If conditions are fulfilled the quote is automatically accepted.
```
Comment (lexerom) [inline openapi/components/schemas/StorefrontQuoteReactivateOrder.yaml:27]:
```suggestion
    description: Conditions must be met for the quote to be automatically accepted.
```

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

**Review by swilczynski — DISMISSED** [review]

File: openapi/components/schemas/StorefrontSubscription.yaml [hunk 49]
```diff
@@ -417,3 +417,12 @@ properties:
...
   _links:
     $ref: ./SelfLink.yaml
+  _embedded:
+    type: object
+    description: Embedded objects that are requested by the `expand` query parameter.
+    readOnly: true
+    properties:
+      changeQuote:
+        type: object
+      reactivateQuote:
```
Comment (swilczynski) [inline openapi/components/schemas/StorefrontSubscription.yaml:427]:
```suggestion
      reactivationQuote:
```

File: openapi/components/schemas/StorefrontSubscription.yaml [hunk 50]
```diff
@@ -417,3 +417,12 @@ properties:
     $ref: ./UpdatedTime.yaml
   _links:
     $ref: ./SelfLink.yaml
+  _embedded:
+    type: object
+    description: Embedded objects that are requested by the `expand` query parameter.
+    readOnly: true
+    properties:
+      changeQuote:
```
Comment (swilczynski) [inline openapi/components/schemas/StorefrontSubscription.yaml:425]:
```suggestion
      creationQuote:
        type: object
       changeQuote:
        type: object
```

**Review by swilczynski — DISMISSED** [review]

**Review by noon-dawg — DISMISSED** [review]

**Review by noon-dawg — DISMISSED** [review]

**Review by danio1024 — DISMISSED** [review]

**Review by swilczynski — DISMISSED** [review]

**Review by XaosSintez — DISMISSED** [review]

**Review by lucasxciv — APPROVED** [review]

**Review by XaosSintez — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**